geoh5py/groups/group.py

```diff
@@ -73,6 +75,28 @@ def add_comment(self, comment: str, author: str = None):
         else:
             self.comments.values = self.comments.values + [comment_dict]
 
+    def clip_by_extent(
+        self, bounds: np.ndarray, parent=None, copy_children: bool = True
+    ) -> Group | None:
+        """
+        Find indices of vertices within a rectangular bounds.
+
+        :param bounds: shape(2, 2) Bounding box defined by the South-West and
+            North-East coordinates. Extents can also be provided as 3D coordinates
+            with shape(2, 3) defining the top and bottom limits.
+        :param attributes: Dictionary of attributes to clip by extent.
+        """
+        new_entity = self.copy(parent=parent, copy_children=False)
```

Comment:
**suggestion (test):** add tests for the cases pointed below by codecov

Reply:
Done